synapse/tools/infra/gendc.py

```diff
@@ -0,0 +1,392 @@
+                target_svc = v.split('_', 1)[1]
+                base_url = svc2ahaconnect[target_svc]
+                url = base_url.format(user=ahaadmin_raw)
+                logger.info(f'Replaced {k}={v} with {url}')
+                conf[k] = url
+                continue
+
+            if isinstance(v, str) and v.startswith('GENSVCAHAURL_'):
+                target_svc = v.split('_', 1)[1]
+                base_url = svc2ahaconnect[target_svc]
+                url = base_url.format(user=svcname)
+                usergens.append((base_url.format(user=ahaadmin_raw), svcuser))
+                logger.info(f'Replaced {k}={v} with {url}')
+                conf[k] = url
+                continue
+
+            if isinstance(v, str) and v.startswith('GENFQDN_'):
+                prefix = v.split('_', 1)[1]
+                fqdn = f'{prefix}.{ahanetwork}'
+                logger.info(f'Replaced {k}={v} with {fqdn}')
+                conf[k] = fqdn
+                continue
+
+        s_common.yamlsave(conf, svcstorage, 'cell.yaml')
+
+        # Generate docker-compose file
+
+        svcdnfo = svcinfo.get('docker')
+
+        svcenv = svcdnfo.get('environment')
+        svcenv.setdefault('SYN_LOG_LEVEL', 'DEBUG')
+        svcenv.setdefault('SYN_LOG_STRUCT', '1')
+
+        svclabels = {k: v for k, v in default_docker_labels.items()}
+        svclabels.update(svcdnfo.get('labels'))
+        svccompose = {'environment': svcenv,
+                      'image': svcdnfo['image'],
+                      'logging': docker_logging,
+                      'network_mode': docker_network_mode,
+                      'restart': docker_restart,
+                      'volumes': ['./storage:/vertex/storage',
+                                  './backups:/vertex/backups']}
+        if svclabels:
+            svccompose['labels'] = svclabels
+
+        svcdc = {
+            'services': {
+                svcfull: svccompose,
+            },
+            'version': '3.3'}
+        s_common.yamlsave(svcdc, svcroot, 'docker-compose.yaml')
+
+        if svcinfo.get('stormsvc', False):
+            stormstr = f'service.add {svcname} {svc_aha_connect.format(user=ahaadmin_raw)}'
+            stormsvcs.append(stormstr)
+
+    logger.info('Copying certdir to ahasvcdir')
+    shutil.copytree(certdirn, s_common.genpath(ahasvcdir, 'certs'), )
+
+    telefp = s_common.genpath(syndir, 'telepath.yaml')
+    logger.info(f'Creating telepath.yaml at {telefp}')
+    tnfo = {
+        'version': 1,
+        'aha:servers': [
+            aharegistry,
+        ]
+    }
+    s_common.yamlsave(tnfo, telefp)
+
+    if stormsvcs:
+        stormfp = s_common.genpath(output_path, 'storm_services.storm')
+        logger.info(f'Saving stormservice config to {stormfp}')
+
+        storm = ' | '.join(stormsvcs)
+        with s_common.genfile(stormfp) as fd:
+            fd.write(storm.encode())
+
+    if usergens:
+        fp = s_common.genpath(output_path, 'usergens.sh')
```

Comment:
I don't really like the concept of plumbing in extra users / vars on boot, since its really a one-time thing and not a persistent todo. this is also a extra knob to enable that "i want to do service specific accounts" that some folks may want.